src/animation/interpolants/SmoothInterpolant.js

```diff
@@ -0,0 +1,103 @@
+/**
+ *
+ * Cubic hermite spline interpolant.
+ *
+ *
+ * @author tschw
+ */
+
+THREE.SmoothInterpolant = function( times, values, stride, result ) {
+
+	THREE.Interpolant.call( this, times, values, stride, result );
+
+};
+
+Object.assign( THREE.SmoothInterpolant.prototype, THREE.Interpolant.prototype, {
+
+	constructor: THREE.SmoothInterpolant,
+
+	DefaultParameters: {
```

Comment:
> I don't remember the names of the inventors, but these are the Lightwave / Max splines :-).

Both Lightwave and Max support TBC splines as defined here:

https://en.wikipedia.org/wiki/Kochanek%E2%80%93Bartels_spline

http://knowledge.autodesk.com/support/3ds-max/learn-explore/caas/CloudHelp/cloudhelp/2015/ENU/3DSMax/files/GUID-40C759C5-80AA-458B-BB57-145A78D8907F-htm.html

> The values you are referring to are Boolean ones, that specify how the spline behaves at the ends when running out of keyframes. 

I thought you could do all that with just additional keys -- you can add extra keys at the beginning and the end.  I always though artists wanted to control all this through setting keys.  I guess this is an implementation choice?  Is there a reference design you are following?